temporal-sdk/src/main/java/io/temporal/client/schedules/ScheduleCalendarSpec.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client.schedules;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Specification relative to calendar time when to run an action.
+ *
+ * <p>A timestamp matches if at least one range of each field matches except for year. If year is
+ * missing, that means all years match. For all fields besides year, at least one range must be
+ * present to match anything.
+ */
+public final class ScheduleCalendarSpec {
+  public static ScheduleCalendarSpec.Builder newBuilder() {
+    return new ScheduleCalendarSpec.Builder();
+  }
+
+  public static ScheduleCalendarSpec.Builder newBuilder(ScheduleCalendarSpec spec) {
+    return new ScheduleCalendarSpec.Builder(spec);
+  }
+
+  public static ScheduleCalendarSpec getDefaultInstance() {
```

Comment:
I added them because almost all other builders in the SDK have them. That being said personally I'd remove almost all of them because I don't see the value either

Reply:
:+1: Yeah, default instances don't mean anything for these I don't think.